Source: Debian, http://patch-tracking.debian.net/patch/series/view/amarok/2.0.60+svn935422-2/03_mysqle_link_to_amarok_executable.diff
        Slightly altered to work on Exherbo and work with upstream modifications
Upstream: no
Reason: Link libmysqle statically in /usr/bin/amarok. This works on all arches, without the need for a shared libmysqle, or -fPIC hackery

commit 445d3cb247584de3cd4f2ea370a834e26c5fd008
Author: Johannes Nixdorf <mixi@user-helfen-usern.de>
Date:   Thu Nov 25 11:14:27 2010 +0100

    Link libmysqle statically

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 3a0ac94..999c4ac 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -959,6 +959,19 @@ if(Q_WS_MAC)
 else(Q_WS_MAC)
     kde4_add_executable(amarok ${amarok_SRCS})
     target_link_libraries(amarok ${KDE4_KDEUI_LIBS} amarokcore amaroklib ${TAGLIB_LIBRARIES} ${TAGLIB-EXTRAS_LIBRARIES} )
+    # Link MySQLe to the executable. Hence -fPIC for libmysqld is not
+    # needed anymore.
+
+    STRING(REPLACE "-Wl,--no-undefined" "" CMAKE_MODULE_LINKER_FLAGS_ALLOW_UNDEFINED ${CMAKE_MODULE_LINKER_FLAGS})
+    SET(CMAKE_MODULE_LINKER_FLAGS ${CMAKE_MODULE_LINKER_FLAGS_ALLOW_UNDEFINED} )
+
+    target_link_libraries(amarok -Wl,-whole-archive ${MYSQL_EMBEDDED_LIBRARIES} -Wl,-no-whole-archive)
+    if(NOT WIN32)
+        if(NOT APPLE)
+            target_link_libraries(amarok crypt pthread dl ssl)
+        endif(NOT APPLE)
+    endif(NOT WIN32)
+    set_target_properties(amarok PROPERTIES LINK_FLAGS "-Wl,--allow-multiple-definition")
 
     install(TARGETS amarok ${INSTALL_TARGETS_DEFAULT_ARGS})
 endif(Q_WS_MAC)
diff --git a/src/core-impl/collections/db/sql/mysqlecollection/CMakeLists.txt b/src/core-impl/collections/db/sql/mysqlecollection/CMakeLists.txt
index aa3588c..872051f 100644
--- a/src/core-impl/collections/db/sql/mysqlecollection/CMakeLists.txt
+++ b/src/core-impl/collections/db/sql/mysqlecollection/CMakeLists.txt
@@ -27,6 +27,9 @@ SET(CMAKE_SHARED_LINKER_FLAGS ${CMAKE_SHARED_LINKER_FLAGS_NOFATALWARN} )
 STRING(REPLACE "-Wl,--fatal-warnings" "" CMAKE_MODULE_LINKER_FLAGS_NOFATALWARN ${CMAKE_MODULE_LINKER_FLAGS})
 SET(CMAKE_MODULE_LINKER_FLAGS ${CMAKE_MODULE_LINKER_FLAGS_NOFATALWARN} )
 
+STRING(REPLACE "-Wl,--no-undefined" "" CMAKE_MODULE_LINKER_FLAGS_ALLOW_UNDEFINED ${CMAKE_MODULE_LINKER_FLAGS})
+SET(CMAKE_MODULE_LINKER_FLAGS ${CMAKE_MODULE_LINKER_FLAGS_ALLOW_UNDEFINED} )
+
 target_link_libraries(amarok_collection-mysqlecollection
     amarok-sqlcollection
     amarokcore
@@ -34,9 +37,6 @@ target_link_libraries(amarok_collection-mysqlecollection
     ${KDE4_KDECORE_LIBS}
     ${QT_QTCORE_LIBRARY}
     ${QT_QTGUI_LIBRARY}
-    ${MYSQL_EMBEDDED_LIBRARIES}
-    ${CMAKE_DL_LIBS}
-    ${ZLIB_LIBRARIES}
 )
 
 if(NOT WIN32 AND NOT APPLE)
